mods/fire/init.lua

```diff
@@ -103,6 +114,20 @@ minetest.register_craft({
 	}
 })
 
+
+-- Override coalblock to enable permanent flame above
+-- Coalblock is non-flammable to avoid unwanted basic_flame nodes
+
+minetest.override_item("default:coalblock", {
+	after_destruct = function(pos, oldnode)
+		pos.y = pos.y + 1
+		if minetest.get_node(pos).name == "fire:permanent_flame" then
+			minetest.remove_node(pos)
+		end
+	end,
```

Comment:
@rubenwardy, suppose someone later wants to add another callback after that one.

With comma:

```
+asdf = ...
```
 
Without:

```
-end
+end,
+asdf = ...
```